user/super/com/google/gwt/emul/java/math/BigInteger.java

```diff
@@ -260,16 +260,40 @@ private static void setFromString(BigInteger bi, String val, int radix) {
    * @throws NumberFormatException if the length of {@code val} is zero.
    */
   public BigInteger(byte[] val) {
+    this(val, 0, val.length);
+  }
+
+  /**
+   * Constructs a new {@code BigInteger} from the given two's complement
+   * representation. The most significant byte is the entry at index 0. The most
+   * significant bit of this entry determines the sign of the new {@code
+   * BigInteger} instance. The given array must not be empty.
+   *
+   * @param val two's complement representation of the new {@code BigInteger}.
+   * @throws NullPointerException if {@code val == null}.
+   * @throws NumberFormatException if the length of {@code val} is zero.
+   */
+  public BigInteger(byte[] val, int offset, int length) {
     if (val.length == 0) {
       // math.12=Zero length BigInteger
       throw new NumberFormatException("Zero length BigInteger"); //$NON-NLS-1$
     }
-    if (val[0] < 0) {
+    if (length < 0 || offset < 0 || val.length < 0 || length > val.length - offset) {
+      throw new IndexOutOfBoundsException("Range check failed: offset=" + offset + ", length="
+          + length + ", val.length=" + val.length);
+    }
+    if (length == 0) {
```

Comment:
It surprises me that this is supported - that's effectively the same thing as passing an empty byte array, but sure enough, it makes a difference:
```
jshell> new BigInteger(new byte[]{127, 127, 127, 127}, 0, 0)
$6 ==> 0

jshell> new BigInteger(new byte[]{}, 0, 0)
|  Exception java.lang.NumberFormatException: Zero length BigInteger
|        at BigInteger.<init> (BigInteger.java:315)
|        at (#7:1)
```